src/uharfbuzz/_harfbuzz.pyx

```diff
@@ -162,17 +168,46 @@ cdef class Buffer:
         if hb_codepoints is not NULL:
             free(hb_codepoints)
 
+    def add_utf8(self, text: bytes,
```

Comment:
Why bytes here?

Reply:
what else should it be?

Reply:
it can't be str (unicode) because the add_utf8 method is intented to be used with UTF-8 encoded bytes. A python unicode string by definition doesn't have any specific "encoding".

If you want I can use a cython memoryview like `const uint8_t[:]` so that it works with any object that supports the buffer interface, bytes, bytearray, array.array, etc.